internal/providers/pluginfw/resources/sharing/resource_share.go

```diff
@@ -0,0 +1,396 @@
+	sharing_tf.ShareInfo
+}
+
+func matchOrder[T any, K comparable](target, reference []T, keyFunc func(T) K) {
+	// Create a map to store the index positions of each key in the reference slice.
+	orderMap := make(map[K]int)
+	for index, item := range reference {
+		orderMap[keyFunc(item)] = index
+	}
+
+	// Sort the target slice based on the order defined in the orderMap.
+	sort.Slice(target, func(i, j int) bool {
+		return orderMap[keyFunc(target[i])] < orderMap[keyFunc(target[j])]
+	})
+}
+
+func suppressCDFEnabledDiff(si *sharing.ShareInfo) {
+	//suppress diff for CDF Enabled if HistoryDataSharingStatus is enabled , as API does not accept both fields to be set
+	for i := range si.Objects {
+		if si.Objects[i].HistoryDataSharingStatus == "ENABLED" {
+			si.Objects[i].CdfEnabled = false
+		}
+	}
+}
+
+func resourceShareMap(si sharing.ShareInfo) map[string]sharing.SharedDataObject {
+	m := make(map[string]sharing.SharedDataObject, len(si.Objects))
+	for _, sdo := range si.Objects {
+		m[sdo.Name] = sdo
+	}
+	return m
+}
+
+func equal(this sharing.SharedDataObject, other sharing.SharedDataObject) bool {
+	if other.SharedAs == "" {
+		other.SharedAs = this.SharedAs
+	}
+	//don't compare computed fields
+	other.AddedAt = this.AddedAt
+	other.AddedBy = this.AddedBy
+	other.Status = this.Status
+	other.HistoryDataSharingStatus = this.HistoryDataSharingStatus
+	other.ForceSendFields = this.ForceSendFields // TODO: is this the right thing to do?
+	return reflect.DeepEqual(this, other)
+}
+
+func diff(beforeSi sharing.ShareInfo, afterSi sharing.ShareInfo) []sharing.SharedDataObjectUpdate {
+	beforeMap := resourceShareMap(beforeSi)
+	afterMap := resourceShareMap(afterSi)
+	changes := []sharing.SharedDataObjectUpdate{}
+	// not in after so remove
+	for _, beforeSdo := range beforeSi.Objects {
+		_, exists := afterMap[beforeSdo.Name]
+		if exists {
+			continue
+		}
+		changes = append(changes, sharing.SharedDataObjectUpdate{
+			Action:     sharing.SharedDataObjectUpdateActionRemove,
+			DataObject: &beforeSdo,
+		})
+	}
+
+	// not in before so add
+	// if in before but diff then update
+	for _, afterSdo := range afterSi.Objects {
+		beforeSdo, exists := beforeMap[afterSdo.Name]
+		if exists {
+			if !equal(beforeSdo, afterSdo) {
+				// do not send SharedAs
+				afterSdo.SharedAs = ""
+				changes = append(changes, sharing.SharedDataObjectUpdate{
+					Action:     sharing.SharedDataObjectUpdateActionUpdate,
+					DataObject: &afterSdo,
+				})
+			}
+			continue
+		}
+		changes = append(changes, sharing.SharedDataObjectUpdate{
+			Action:     sharing.SharedDataObjectUpdateActionAdd,
+			DataObject: &afterSdo,
+		})
+	}
+	return changes
+}
+
+func shareChanges(si sharing.ShareInfo, action string) sharing.UpdateShare {
+	var changes []sharing.SharedDataObjectUpdate
+	for _, obj := range si.Objects {
+		changes = append(changes, sharing.SharedDataObjectUpdate{
+			Action:     sharing.SharedDataObjectUpdateAction(action),
+			DataObject: &obj,
+		},
+		)
```

Comment:
Curious, did this pass `gofmt`?
```suggestion
		changes = append(changes, sharing.SharedDataObjectUpdate{
			Action:     sharing.SharedDataObjectUpdateAction(action),
			DataObject: &obj,
		})
```

Reply:
with flying colors 🌈